src/dom_types/values.rs

```diff
@@ -25,14 +25,47 @@ impl<T: ToString> From<T> for CSSValue {
     }
 }
 
-// `&` because `style!` macro automatically adds prefix `&` before values for more ergonomic API
-// (otherwise it would fail when you use for example a Model's property in View functions as `CSSValue`)
-impl From<&CSSValue> for CSSValue {
-    fn from(value: &CSSValue) -> Self {
-        value.clone()
+// ----------- ToCssValue impls ------------
+
+// impl ToCssValue for CSSValue
+#[doc(hidden)]
+pub trait ToCssValue {
+    fn to_css_value(self) -> CSSValue;
+}
+
+impl ToCssValue for CSSValue {
+    fn to_css_value(self) -> CSSValue {
+        self
     }
 }
 
+// impl<T: ToString> ToCssValue for T
+#[doc(hidden)]
+pub trait ToCssValueForToString {
+    fn to_css_value(&self) -> CSSValue;
+}
+
+impl<T: ToString> ToCssValueForToString for T {
+    fn to_css_value(&self) -> CSSValue {
+        CSSValue::Some(self.to_string())
+    }
+}
+
+// impl<T: ToString> ToCssValue for Option<T>
+#[doc(hidden)]
+pub trait ToCssValueForOptionToString {
+    fn to_css_value(&self) -> CSSValue;
+}
+
+impl<T: ToString> ToCssValueForOptionToString for Option<T> {
+    fn to_css_value(&self) -> CSSValue {
+        self.as_ref()
+            .map_or(CSSValue::Ignored, |t| CSSValue::Some(t.to_string()))
+    }
+}
+
+// TODO: Should we impl ToCssValue for more types? such as Vec<T>, Box<T>.. etc
```

Comment:
I think we can remove it now.

Reply:
Remove the `TODO` comment ?

Reply:
Yeah, it was good for code review, but I think we don't need it anymore.

Reply:
Well, I didn't put it for this review only, I put it as reminder for the future as well, but I will remove it since it's not hard to remember :D

Reply:
Yeah, thanks, there are too many TODOs already :)